sys/kern/src/arch/riscv32.rs

```diff
@@ -435,11 +435,17 @@ fn timer_handler() {
                 set_current_task(next);
             }
 
-            // Reset mtime back to 0.  In theory we could save an instruction on
-            // RV32 here and only write the low-order bits, assuming that it has
-            // been less than 12 seconds or so since our last interrupt(!), but
-            // let's avoid any possibility of a nasty surprise.
-            core::ptr::write_volatile(crate::startup::MTIME as *mut u64, 0);
+            //
+            // Increase mtimecmp for the next interrupt
```

Comment:
Since you are cleaning this up, I think we should move this new code that you added to `set_timer` and drop the `asm!` code in it. I don't know why we have special asm usage to set `mtime` and `mtimecmp` in `set_timer`. In fact, we don't even need to set mtime to 0. We can just start with whatever the hardware reset default is and always call `set_timer` during init and for every timer interrupt.

Reply:
Agreed, went ahead and moved some stuff around.